docs/docs/reference/proxy.md

```diff
@@ -0,0 +1,147 @@
+---
+sidebar_label: Proxy
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Proxy: Authenticate and make API calls
+
+:::tip
+Make sure you have gone through the [quickstart](/quickstart) before using the proxy!
+:::
+
+# What Is It?
+Same familiar simple interface with a lot of functionality baked in:
+1. Handles authentication
+2. Handles retries
+3. Takes care of logging
+4. Handles rate limits
+5. Scales as you scale
+
+```
+const repsonse = await nango.proxy({
+    endpoint: '/user',
+    providerConfigKey: 'XXX',
+    connectionId: 'XXX'
+});
+```
+
+## SDK Usage
+
+1. Install and set up the SDK as described in [the sdk docs](/reference/node-sdk)
+2. Use the proxy to both authenticate and make an API call in one simple step
+```js
+# note these params closely follows how axios structures their API
+const response = await nango.proxy({
+    method: '<HTTP-METHOD>',    // GET is the default if not provided
+    endpoint: '<API-ENDPOINT>', // exclude the base API URL
+    providerConfigKey: '<PROVIDER-CONFIG-KEY>',
+    connectionId: '<CONNECTION-ID>',
+
+    // optional parameters
+    headers: '<HEADERS>',
+    params: '<PARAMS>',   // URL parameters to be sent with the request. Must be a plain object or a URLSearchParams object
+    paramsSerializer: {
+        encode?: (param: string): string => { /* Do custom ops here and return transformed string */ }, // custom encoder function; sends Key/Values in an iterative fashion
+        serialize?: (params: Record<string, any>, options?: ParamsSerializerOptions ), // mimic pre 1.x behavior and send entire params object to a custom serializer func. Allows consumer to control how params are serialized.
+        indexes: false // array indexes format (null - no brackets, false (default) - empty brackets, true - brackets with indexes)
+      },
+    data: {
+        param: 'VALUE'
+    }
+});
+```
+
+### Example
+```js
+const insertCalendarListResponse = await nango.proxy({
+    method: 'POST',
+    endpoint: '/users/me/calendarList',
+    providerConfigKey: '<PROVIDER-CONFIG-KEY>',
+    connectionId: '<CONNECTION-ID>',
+    data: {
+        id: 1,
+        colorId: 'blue',
+        selected: true
+    }
+});
+```
+
+## REST API Usage
+
+<Tabs groupId="deployment" queryString>
+  <TabItem value="cloud" label="Nango Cloud">
+
+```bash
+curl -H 'Authorization: Bearer <SECRET-KEY>' \
```

Comment:
I was wondering if it would not be better to stay as close to the original request structure as possible. 

For example here: 
`https://api.nango.dev/proxy/API/ENDPOINT?param1=x&param2=y&nango_connection_id=<CONNECTION-ID>&nango_provider_config_key=<CONFIG-KEY>`

In this case, we would pick up the path as well as param1/param2 and use this to make the request. We could have similar logic to automatically forward the headers and body. 

This would allow us to reduce the amount of code to change to adop the proxy, and just minimize the deviance to making a direct HTTP request.

We could even ask to pass in the connection ID and unique key as `nango`-prefixed headers, if we want the request to look as unaltered as possible.

Reply:
Interesting thought. I'll play with this some and see if it plays nicely

Reply:
Done with https://github.com/NangoHQ/nango/pull/562/commits/16b84d2e057104252d926f52c10e216be35ab95a

The connection id and config key are now passed in the header:

```
curl -X PATCH \
-H 'Connection-Id: github-k1' \
-H 'Provider-Config-Key: github' \
-d '{"company": "NangoHQ"}' \
'http://localhost:3003/proxy/user'
```